src/GafferCycles/IECoreCyclesPreview/Renderer.cpp

```diff
@@ -1969,6 +1996,12 @@ class InstanceCache : public IECore::RefCounted
 
 	private :
 
+		/// \todo Figure out if this can be refactored to return `ccl::Geometry`
```

Comment:
Long long ago from a memory now almost forgotten, the ccl::Geometry base class didn't exist which made supporting things like volumes and curves a bit tricky without going the ccl::Object approach instead. I could be wrong but yeah this can be considered for refactoring now. (Not related to here, but I think Cycles really needs a real instance object of sorts that stores a list of geometries/prototypes as a list + some pointcloud of xforms/constants).

Reply:
Cool - thanks for the insights. I'll hopefully take a stab at this at some point...